design/baremetal-operator/bmh_live_boot.md

```diff
@@ -0,0 +1,121 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Add boot-iso API to BareMetalHost
+
+## Status
+
+implementable
+
+## Summary
+
+Add a new interface so that it is possible to boot arbitrary
+iso images via Ironic.
+
+## Motivation
+
+In some circumstances it is desirable to boot an existing iso image
+instead of having Ironic boot IPA, for example:
+
+To reduce boot time for ephemeral workloads, it may be possible to boot an iso
+and not deploy any image to disk (saving the time to write the image and reboot)
+
+Where an alternative installer exists that is distributed as a live-iso, it
+may be desirable to leverage that toolchain instead of the IPA deploy ramdisk,
+for example the [fedora-coreos installer](https://docs.fedoraproject.org/en-US/fedora-coreos/bare-metal/)
+
+An [Ironic API](https://docs.openstack.org/ironic/latest/admin/ramdisk-boot.html)
+exists that supports this but it's not currently accessible via metal3.
+
+### Goals
+
+Expose the Ironic API to boot iso images via metal3.
+
+### Non-Goals
+
+This only considers the ability to boot an iso image (which Ironic supports
+via both redfish-virtualmedia and iPXE), not live kernel/ramdisk images.
+
+Ironic does support booting kernel/ramdisk images though, so the interface
+we decide on shouldn't preclude adding that capability in future.
+
+## Proposal
+
+Add a new field to the BareMetalHost spec which indicates that instead of
+deploying an image to disk, a live-iso image will be booted:
+
+```yaml
+  spec:
+    liveImage:
+      url: http://1.2.3.4/image.iso
+    online: true
+```
+
+Note that in this mode, `rootDeviceHints` and `userData` will not be used
+since the image won't be written to disk, and Ironic doesn't
+support passing user-data in addition to the iso attachment at present.
```

Comment:
It's about time we added a webhook to validate this kind of stuff.

Reply:
There's a period of time when validation web hooks aren't applied, between the time the CRD is installed (and the API server supports the new resource type) and when the web hook server is registered and starts.